hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/WALEntryStream.java

```diff
@@ -470,7 +470,27 @@ private Pair<WALTailingReader.State, Boolean> readNextEntryAndRecordReaderPositi
     // in the future, if we want to optimize the logic here, for example, do not call this method
     // every time, or do not acquire rollWriteLock in the implementation of this method, we need to
     // carefully review the optimized implementation
```

Comment:
nit: now that we have comments for both if and else, maybe good to remove from here?